### PR TITLE
Gracefully handling invalid Graphite connection.

### PR DESCRIPTION
**Pull request by kovacshuni**

### CODE REVIEW

**Comment by Zapadlo** [comment]
+1

File: graphitefeeder.go [hunk 1]
```diff
@@ -52,7 +52,9 @@ func (graphite *GraphiteFeeder) sendBuffer(bufferGraphite chan *HealthTimed) err
 
 func (graphite *GraphiteFeeder) sendPilotLight() error {
 	if graphite.connection == nil {
-		return errors.New("No graphite connection")
+		msg := "WARN Can't send pilot light, no Graphite connection."
+		log.Printf(msg)
+		return errors.New(msg)
```
Comment (martingartonft) [inline graphitefeeder.go]: seems wrong to log here.  why not get the caller to log?

File: graphitefeeder.go [hunk 2]
```diff
@@ -64,6 +66,11 @@ func (graphite *GraphiteFeeder) sendPilotLight() error {
 }
 
 func (graphite *GraphiteFeeder) sendOne(result *HealthTimed) error {
+	if graphite.connection == nil {
+		msg := "WARN Can't send results, no Graphite connection."
+		log.Printf(msg)
+		return errors.New(msg)
+	}
```
Comment (martingartonft) [inline graphitefeeder.go]: same here.  also you're using printf with no formatting.  fmt.Println() would be better.

File: graphitefeeder.go [hunk 3]
```diff
@@ -88,14 +95,15 @@ func addBack(bufferGraphite chan<- *HealthTimed, healthTimed *HealthTimed) {
...
+	if graphite.connection != nil {
+		graphite.connection.Close()
+	}
+	graphite.connection = tcpConnect(graphite.host, graphite.port)
 }
 
 func tcpConnect(host string, port int) net.Conn {
 	conn, err := net.Dial("tcp", host+":"+strconv.Itoa(port))
-	if err != nil {
+	if conn == nil || err != nil {
```
Comment (martingartonft) [inline graphitefeeder.go]: now can conn be nil and also err be nil?

Comment (kovacshuni) [inline graphitefeeder.go]: removed.

File: graphitefeeder.go (hunk 1 above)
Comment (kovacshuni) [inline graphitefeeder.go]: Lifted the logging. Is it ok?

**Comment by martingartonft** [comment]
lgtm

**Pull request merged**